src/pybind/mgr/dashboard/services/nvmeof_conf.py

```diff
@@ -82,16 +92,33 @@ def get_service_info(cls, group=None):
             if not gateways:
                 return None
 
-            if group:
-                for service_name, entries in gateways.items():
-                    if group in service_name:
-                        entry = next((entry for entry in entries if entry['group'] == group), None)
-                        if entry['group'] == group:  # type: ignore
-                            return service_name, entry['service_url']  # type: ignore
+            try:
+                if group:
+                    orch = OrchClient.instance()
+
+                    for service_name, entries in gateways.items():
+                        # nvmeof naming convention is nvmeof.<rbd-pool>.<group>
+                        # so taking the 3rd entry from the split
+                        if group == service_name.split('.')[2]:
+                            daemons = [d.to_dict() for d in orch.services.list_daemons(
+                                service_name=service_name)]
+
+                            # get the running nvmeof daemons
+                            daemon_names = [d['daemon_name'] for d in daemons
+                                            if d['status_desc'] == 'running']
+                            entry = next((entry for entry in entries
+                                          if entry['daemon_name'] in daemon_names), None)
```

Comment:
I've seen this pattern so many times repeated in Cephadm & Dashboard code, that I think we should extend some orch interfaces to support, for example, returning only the daemons that are running:
```py
orch.services.list_daemons(service_name=<whatever>, status=['running'])
```
In fact, this whole method seems to be to obtain the information of a specific service instance, so something feels wrong here.

Reply:
i opened a separate PR and doing it [in this commit](https://github.com/ceph/ceph/pull/59626/commits/e32d225171f1ef1983200123b62f2f7d02763a6f). It feels out of the scope for this PR so probably we can enhance it there?